Functional.AsyncEnumerables/IAsyncEnumerator.cs

```diff
@@ -1,14 +1,12 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Text;
 using System.Threading.Tasks;
 
 namespace Functional
 {
-    public interface IAsyncEnumerator<out T>
-    {
+	public interface IAsyncEnumerator<out T> : IDisposable
```

Comment:
I hesitant to do this, and I think I'd prefer to wait until .NET Standard 2.1 comes out so I can delete these types altogether.

Reply:
This is suggested by IDisposableAnalyzers,  to make it align with IEnumerator which implements IDisposable as well